src/VisualStudio/Core/Def/Telemetry/AggregatingTelemetryLog.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.VisualStudio.Telemetry;
+using Microsoft.VisualStudio.Telemetry.Metrics;
+using Microsoft.VisualStudio.Telemetry.Metrics.Events;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Telemetry
+{
+    /// <summary>
+    /// Provides a wrapper around the VSTelemetry histogram APIs to support aggregated telemetry. Each instance
+    /// of this class corresponds to a specific FunctionId operation and can support aggregated values for each
+    /// metric name logged.
+    /// </summary>
+    internal sealed class AggregatingTelemetryLog : ITelemetryLog
+    {
+        private readonly IMeter _meter;
+        private readonly TelemetrySession _session;
+        private readonly Dictionary<string, IHistogram<int>> _histograms;
+        private readonly object _lock;
```

Comment:
doc what thsi locks.

Reply:
if it's just the dictionary, move both to bottom after all the non-changing state.